src/test/java/bio/terra/service/profile/ProfileAPIControllerTest.java

```diff
@@ -141,44 +150,66 @@ public void testUpdateProfileForbidden() {
     verify(profileService, times(0)).updateProfile(eq(billingProfileUpdateModel), eq(user));
   }
 
-  @Test
-  public void testDeleteProfile() {
+  @ParameterizedTest
+  @MethodSource
+  void testDeleteProfile(boolean deleteCloudResources, int expectedAdminAuthNumberOfInvocations) {
     when(authenticatedUserRequestFactory.from(any())).thenReturn(user);
     UUID deleteId = UUID.fromString("aaaaaaaa-bbbb-cccc-dddd-eeeeeeeeeeee");
     String jobId = "jobId";
-    when(profileService.deleteProfile(eq(deleteId), eq(user))).thenReturn(jobId);
+    when(profileService.deleteProfile(eq(deleteId), eq(deleteCloudResources), eq(user)))
+        .thenReturn(jobId);
+    when(profileService.getProfileByIdNoCheck(deleteId))
+        .thenReturn(new BillingProfileModel().id(deleteId));
+    doNothing().when(iamService).verifyAuthorization(any(), any(), any(), any());
 
     var jobModel = new JobModel();
     jobModel.setJobStatus(JobStatusEnum.RUNNING);
     when(jobService.retrieveJob(eq(jobId), eq(user))).thenReturn(jobModel);
 
-    ResponseEntity entity = apiController.deleteProfile(deleteId);
-    verify(profileService, times(1)).deleteProfile(eq(deleteId), eq(user));
+    ResponseEntity entity = apiController.deleteProfile(deleteId, deleteCloudResources);
+    // Only check for admin auth if deleteCloudResources is true
+    verify(iamService, times(expectedAdminAuthNumberOfInvocations))
+        .verifyAuthorization(
+            eq(user), eq(IamResourceType.DATAREPO), any(), eq(IamAction.CONFIGURE));
+    // Always check that the user has access on the spend profile
+    verify(iamService, times(1))
+        .verifyAuthorization(
+            eq(user),
+            eq(IamResourceType.SPEND_PROFILE),
+            eq(deleteId.toString()),
+            eq(IamAction.DELETE));
+    verify(profileService, times(1))
+        .deleteProfile(eq(deleteId), eq(deleteCloudResources), eq(user));
     assertNotNull(entity);
   }
 
+  private static Stream<Arguments> testDeleteProfile() {
+    return Stream.of(arguments(true, 1), arguments(false, 0));
+  }
+
   @Test
-  public void testDeleteProfileNotFound() {
+  void testDeleteProfileNotFound() {
     UUID profileId = UUID.randomUUID();
     doThrow(ProfileNotFoundException.class).when(profileService).getProfileByIdNoCheck(profileId);
-    assertThrows(ProfileNotFoundException.class, () -> apiController.deleteProfile(profileId));
+    assertThrows(
+        ProfileNotFoundException.class, () -> apiController.deleteProfile(profileId, false));
     verifyNoInteractions(iamService);
-    verify(profileService, times(0)).deleteProfile(eq(profileId), eq(user));
+    verify(profileService, times(0)).deleteProfile(eq(profileId), eq(false), eq(user));
```

Comment:
Instead of `times(0)` you can write `never()`. Also these are all literal arguments, so `eq` can be removed.
```suggestion
    verify(profileService, never()).deleteProfile(profileId, false, user);
```